### PR TITLE
TST: Skip test_database_uri_string if pg8000 importable

### LINKED ISSUE #16527: TestSQLApi/TestSQLApiConn::test_database_uri_string errors if pg8000 is installed
#### Code Sample, a copy-pastable example if possible

```python
db_uri = "postgresql+pg8000://user:pass@host/dbname"
with tm.assert_raises_regex(ImportError, "pg8000"):
    sql.read_sql("select * from table", db_uri)
```
#### Problem description

The test code comments specify that `pg8000` is a "driver that will not be installed on Travis". However, when the test suite is run elsewhere, that driver may be installed.

#### Expected Output

The test should pass or skip in any environment where pandas is expected to run successfully.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: FreeBSD
OS-release: 10.3-STABLE
machine: amd64
processor: amd64
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.1
pytest: 3.1.0
pip: None
setuptools: 32.1.0
Cython: 0.25.2
numpy: 1.11.2
scipy: 0.19.0
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.0.0
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 1.5.3
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.6
lxml: 3.6.0
bs4: 4.5.1
html5lib: 0.9999999
sqlalchemy: 1.1.10
pymysql: 0.7.11.None
psycopg2: 2.7.1 (dt dec pq3 ext lo64)
jinja2: 2.9.5
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by neirbowj**
- [x] closes #16527
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -938,6 +938,12 @@ def test_database_uri_string(self):
 
         # using driver that will not be installed on Travis to trigger error
         # in sqlalchemy.create_engine -> test passing of this error to user
```
Comment (jreback) [inline pandas/tests/io/test_sql.py]: you would need to do

```
pg8000 = pytest.pyimportskip('pg8000')  # noqa
```

you need to explicitly install this though in the requirements-2.7.pip (and other ones if needed)

cc @jorisvandenbossche

**Review by jreback — CHANGES_REQUESTED** [review]
actually how did this come up? reading the comments suggest that we *don't* want to have this driver installed at all (we are testing an error when sqlalchemy is installed but not a specific driver).

File: pandas/tests/io/test_sql.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/io/test_sql.py]: @neirbowj ahh I see what you posted on the issue. ok then your fix is fine (just need the noqa), and put an explanation of what is going on here.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@jorisvandenbossche ok?

**Comment by TomAugspurger** [comment]
This looks good, thanks.

**Comment by jorisvandenbossche** [comment]
Yes, perfect! Thanks @neirbowj

**Pull request merged**